### PR TITLE
Add standardese

### PR DESCRIPTION
**Pull request by saraedum**
0.4.0 is not out yet, but let's see what needs to happen to make it build here

see https://github.com/standardese/standardese/issues/125

(I'll release 0.4.0 once this works actually.)

### CODE REVIEW

File: recipes/standardese/0002-search-libclang.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/external/external.cmake b/external/external.cmake
+index 8a74eb8..d6b08f6 100644
+--- a/external/cppast/external/external.cmake
++++ b/external/cppast/external/external.cmake
+@@ -176,7 +176,7 @@ function(_cppast_find_libclang config_tool min_version force)
+     if(NOT LIBCLANG_LIBRARY)
+         execute_process(COMMAND ${config_tool} --libdir
+                         OUTPUT_VARIABLE llvm_library_dir OUTPUT_STRIP_TRAILING_WHITESPACE)
+-        find_library(LIBCLANG_LIBRARY "clang" "${llvm_library_dir}" NO_DEFAULT_PATH)
++        find_library(LIBCLANG_LIBRARY "libclang.lib" "${llvm_library_dir}")
```
Comment (isuruf) [inline recipes/standardese/0002-search-libclang.patch]:
```suggestion
+        find_library(LIBCLANG_LIBRARY NAMES clang libclang HINTS "${llvm_library_dir}")
```

**Comment by saraedum** [comment]
@conda-forge/staged-recipes this is ready for review :)

File: recipes/standardese/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  run_exports:
+    # standardese uses semantic versioning
+    - {{ pin_subpackage("standardese") }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - boost
```
Comment (isuruf) [inline recipes/standardese/meta.yaml]:
```suggestion
    - boost-cpp
```

Comment (saraedum) [reply recipes/standardese/meta.yaml]: We need boost ProgramOptions and Filesystem. Afaik the headers are not enough for that.

Comment (saraedum) [reply recipes/standardese/meta.yaml]: (or am I misunderstanding what boost-cpp is?)

Comment (isuruf) [reply recipes/standardese/meta.yaml]: `boost` conda package is `boost-cpp` conda package + boost python libraries.

Comment (saraedum) [reply recipes/standardese/meta.yaml]: Ah, I thought the `cpp` was referring to headers only, like `cpp` the preprocessor. Thanks for clarifying :)

**Comment by saraedum** [comment]
@isuruf, thanks for having a look. I think this is ready now.

**Comment by isuruf** [comment]
Is this linking to boost statically?

**Comment by saraedum** [comment]
It probably is. Is that a problem or should I just take it out of run?

**Comment by isuruf** [comment]
Yes. Try deleting Boost's Cmake files and use CMake's Boost files.

**Comment by saraedum** [comment]
I don't understand what you are suggesting but I guess I can also just change this line:
```
set(Boost_USE_STATIC_LIBS ON)
```

**Comment by isuruf** [comment]
Sure. Let's try that.

**Comment by saraedum** [comment]
This doesn't make sense to me:

```
-- No suitable boost_program_options variant has been identified!
--   libboost_program_options.so.1.70.0 (shared, BUILD_SHARED_LIBS not ON, set Boost_USE_STATIC_LIBS=OFF to override)
--   libboost_program_options.a (static, Boost_USE_STATIC_LIBS=OFF)
```
I'll try with BUILD_SHARED_LIBS=ON. But maybe I understand what you meant by deleting Boost's cmake now.

**Comment by saraedum** [comment]
Seems like this worked…

File: recipes/standardese/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - boost-cpp
+    - clangdev
+  run:
+    - {{ pin_compatible('libclang') }}
+    - {{ pin_compatible('boost-cpp') }}
```
Comment (isuruf) [inline recipes/standardese/meta.yaml]:
```suggestion
    - boost-cpp
```

Global pinning from conda-forge will apply. (Boost breaks ABI in minor versions and globally boost is pinned to 1.70)

Comment (saraedum) [reply recipes/standardese/meta.yaml]: Right…I should have known that.

**Comment by saraedum** [comment]
Thanks a lot for your help @isuruf!

**Pull request merged**